app/controllers/UserController.scala

```diff
@@ -0,0 +1,77 @@
+package controllers
+
+import common.{Admin, Driver}
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import security.UserAction
+import services.UserService
+
+import scala.concurrent.{ExecutionContext, Future}
+import common.Serialization.{INSTANCE => Json}
+
+@Singleton
+class UserController @Inject()(userService: UserService,
+                               cc: MessagesControllerComponents,
+                               override val userAction: UserAction
+                              )(implicit ec: ExecutionContext)
+  extends SecuredController(cc, userAction) {
+
+  def getFiltered(name: String): Action[AnyContent] = securedAsync(Admin.INSTANCE :: Nil, Action.async {
+    Future.successful(NotImplemented(""))
+  })
+
+  def get(name: String): Action[AnyContent] = securedAsync(Admin.INSTANCE :: Nil, Action.async {
+    _ => {
+      userService.get(name).map {
+        case None => NotFound(Json.toJson(new common.BadRequest(s"User with name $name not found")))
+        case Some(user) => Ok(Json.toJson(user))
+      }
+    }
+  })
+
+  def get(id: Int): Action[AnyContent] = securedAsync(Admin.INSTANCE :: Nil, Action.async {
+    _ => {
+      userService.get(id).map {
+        case None => NotFound(Json.toJson(new common.BadRequest(s"User with id $id not found")))
+        case Some(user) => Ok(Json.toJson(user))
+      }
+    }
+  })
+
+  def create(): Action[AnyContent] = securedAsync(Admin.INSTANCE :: Nil, Action.async {
+    request => {
+      //      todo: extract this json processing/bad-requesting
+      val body = request.body.asText
+      body match {
+        case Some(userData) =>
+          val user = Json.fromJson(userData, classOf[common.User])
+          userService.createUser(user).map {
+            user => Ok(Json.toJson(user))
```

Comment:
Wut
You're deserializing and then serializing again?

Reply:
yep, I deserialize content that came in http request body, process it (shove it down `UserService`) and get resulting created user (with id and stuff) and serialize it back to return as http response

Reply:
Jeez